src/configuration.ts

```diff
@@ -177,18 +184,21 @@ function findup(filename: string, directory: string): string | undefined {
     }
 
     function findFile(cwd: string): string | undefined {
-        if (fs.existsSync(path.join(cwd, filename))) {
-            return filename;
-        }
+        for (const filename of filenames) {
+            if (fs.existsSync(path.join(cwd, filename))) {
+                return filename;
+            }
 
-        // TODO: remove in v6.0.0
-        // Try reading in the entire directory and looking for a file with different casing.
-        const filenameLower = filename.toLowerCase();
-        const result = fs.readdirSync(cwd).find((entry) => entry.toLowerCase() === filenameLower);
-        if (result !== undefined) {
-            showWarningOnce(`Using mixed case tslint.json is deprecated. Found: ${path.join(cwd, result)}`);
+            // TODO: remove in v6.0.0
+            // Try reading in the entire directory and looking for a file with different casing.
+            const filenameLower = filename.toLowerCase();
```

Comment:
you can probably remove this line while you are here. we already know filename is lowercase